std/conv.d

```diff
@@ -4946,6 +4946,17 @@ unittest //Constness
     emplaceRef!(IS[2])(ss, iss[]);
 }
 
+unittest
+{
+    struct S
+    {
+        void opAssign(T)(T v) {}
+        Object o = new Object;
```

Comment:
It is a bug that this code is accepted. The initializer is in global memory and is implicitly shared. I can't find the bug report for it right now but I know it is filed. The compiler correctly rejects mutable class variables with compile-time initializers in other contexts, such as for module-level variables and static function variables.

That said, if we do implement compile-time class variable initializers to support mutable types by placing in TLS instead of global memory, there would still be the problem of `emplaceInitializer` not working with it.